components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Models/BaseMonitoredItemModel.cs

```diff
@@ -69,23 +70,51 @@ public class MonitoredItemModel {
         public string TriggerId { get; set; }
 
         /// <summary>
-        /// Data change filter
+        /// Clones this object
         /// </summary>
-        public DataChangeFilterModel DataChangeFilter { get; set; }
+        /// <returns></returns>
+        public abstract BaseMonitoredItemModel Clone();
 
         /// <summary>
-        /// Event filter
+        /// Compare items
         /// </summary>
-        public EventFilterModel EventFilter { get; set; }
-
-        /// <summary>
-        /// Aggregate filter
-        /// </summary>
-        public AggregateFilterModel AggregateFilter { get; set; }
-
-        /// <summary>
-        /// heartbeat interval not present if zero
-        /// </summary>
-        public TimeSpan? HeartbeatInterval { get; set; }
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public virtual bool IsSameAs(BaseMonitoredItemModel other) {
```

Comment:
Any specific reason not to use .NET Standard IComparable or to overload == and != operator?

Reply:
The IsSameAs pattern is used *everywhere* else in the code - I moved this code from from the MonitoredItemModelEx extension into the base class. However, I like your suggestion - I was thinking of implementing IComparable but didn't do. I think overriing Equals() is a better alternative. However, all the sub objects (like DataChangeFilterModel etc) does not follow this pattern right now. Do we think that I should change this for all applicable classes? There will be classes that still does IsSameAs - will it be weird to have two different patterns? That's what I'm thinking about anyway

Reply:
Got it, sounds good I will ask the same questions someone else ;)

Reply:
I just committed a version where we implement Equals in our new classes, but keep IsSameAs everywhere else. Take a look at it. If you want I can revert to IsSameAs in these new classes as well.

Reply:
Looking at the code right now there are 31 implementations of IsSameAs for different classes. So it wouldn't be impossible to refactor them all if we would like to. It does change a lot of code though - so it's a high-risk change.

Reply:
If you see it as high-risk based on the amount of code that needs to be changed. We should leave like it is, for now. Thanks